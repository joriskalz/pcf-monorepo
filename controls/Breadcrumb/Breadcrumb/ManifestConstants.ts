export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
