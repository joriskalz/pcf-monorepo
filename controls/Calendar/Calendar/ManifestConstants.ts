export const enum InputEvents {
    SetFocus = 'SetFocus',
}
