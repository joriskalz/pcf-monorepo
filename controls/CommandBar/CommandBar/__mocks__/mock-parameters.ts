/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
