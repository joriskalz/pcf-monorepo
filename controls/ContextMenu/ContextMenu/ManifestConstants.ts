export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    ItemHeader = 'ItemHeader',
    ItemTopDivider = 'ItemTopDivider',
    ItemDivider = 'ItemDivider',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const TextAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};
