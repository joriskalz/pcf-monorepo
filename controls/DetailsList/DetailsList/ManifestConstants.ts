export const enum RecordsColumns {
    RecordKey = 'RecordKey',
    RecordCanSelect = 'RecordCanSelect',
    RecordSelected = 'RecordSelected',
}

export const enum ColumnsColumns {
    ColDisplayName = 'ColDisplayName',
    ColName = 'ColName',
    ColWidth = 'ColWidth',
    ColSortable = 'ColSortable',
    ColSortBy = 'ColSortBy',
    ColHorizontalAlign = 'ColHorizontalAlign',
    ColVerticalAlign = 'ColVerticalAlign',
    ColMultiLine = 'ColMultiLine',
    ColIsBold = 'ColIsBold',
    ColTagColorColumn = 'ColTagColorColumn',
    ColTagBorderColorColumn = 'ColTagBorderColorColumn',
    ColResizable = 'ColResizable',
    ColHeaderPaddingLeft = 'ColHeaderPaddingLeft',
    ColCellType = 'ColCellType',
    ColShowAsSubTextOf = 'ColShowAsSubTextOf',
    ColPaddingTop = 'ColPaddingTop',
    ColPaddingLeft = 'ColPaddingLeft',
    ColLabelAbove = 'ColLabelAbove',
    ColMultiValueDelimiter = 'ColMultiValueDelimiter',
    ColFirstMultiValueBold = 'ColFirstMultiValueBold',
    ColInlineLabel = 'ColInlineLabel',
    ColHideWhenBlank = 'ColHideWhenBlank',
    ColSubTextRow = 'ColSubTextRow',
    ColAriaTextColumn = 'ColAriaTextColumn',
    ColCellActionDisabledColumn = 'ColCellActionDisabledColumn',
    ColImageWidth = 'ColImageWidth',
    ColImagePadding = 'ColImagePadding',
    ColRowHeader = 'ColRowHeader',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
    SetFocusOnRow = 'SetFocusOnRow',
    SetFocusOnRowSetSelection = 'SetFocusOnRowSetSelection',
    ClearSelection = 'ClearSelection',
    SetSelection = 'SetSelection',
    SetFocusOnHeader = 'SetFocusOnHeader',
    LoadNextPage = 'LoadNextPage',
    LoadPreviousPage = 'LoadPreviousPage',
    LoadFirstPage = 'LoadFirstPage',
}

export const enum OutputEvents {
    Sort = 'Sort',
    CellAction = 'CellAction',
    OnRowSelectionChange = 'OnRowSelectionChange',
}

export const enum SortDirection {
    Ascending = '0',
    Descending = '1',
}

export const enum CellTypes {
    Expand = 'expand',
    Tag = 'tag',
    Image = 'image',
    ClickableImage = 'clickableimage',
    IndicatorTag = 'indicatortag',
    Link = 'link',
}
