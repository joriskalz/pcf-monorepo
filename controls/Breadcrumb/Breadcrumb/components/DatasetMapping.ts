import { IBreadcrumbItem } from '@fluentui/react';
import { ItemColumns } from '../ManifestConstants';
import { ICustomBreadcrumbItem } from './components.types';

export function getItemsFromDataset(dataset: ComponentFramework.PropertyTypes.DataSet): ICustomBreadcrumbItem[] {
    if (dataset.error || dataset.paging.totalResultCount === undefined) {
        // Dataset is not defined so return dummy items
        return [getDummyAction('1'), getDummyAction('2'), getDummyAction('3')];
    }
    const keyIndex: Record<string, number> = {};
    return dataset.sortedRecordIds.map((id) => {
        const record = dataset.records[id];
        // Prevent duplicate keys by appending the duplicate index
        let key = record.getValue(ItemColumns.Key) as string;
        if (keyIndex[key] !== undefined) {
            keyIndex[key]++;
            key += `_${keyIndex[key]}`;
        } else keyIndex[key] = 1;
        return {
            id: record.getRecordId(),
            key: key,
            text: record.getFormattedValue(ItemColumns.DisplayName),
            clickable: record.getValue(ItemColumns.Clickable) as boolean,
            data: record,
        } as ICustomBreadcrumbItem;
    });
}

export function getBreadcrumbItems(
    items: ICustomBreadcrumbItem[],
    onClick: (ev?: React.MouseEvent<HTMLElement>, item?: ICustomBreadcrumbItem) => void,
): IBreadcrumbItem[] {
    return items.map((item) =>
        item.clickable === false
            ? {
                  id: item.id,
                  key: item.key,
                  text: item.text,
              }
            : {
                  id: item.id,
                  key: item.key,
                  text: item.text,
                  onClick: onClick,
              },
    ) as IBreadcrumbItem[];
}

function getDummyAction(key: string): ICustomBreadcrumbItem {
    return {
        id: key,
        key: key,
        text: 'Item ' + key,
        clickable: true,
    } as ICustomBreadcrumbItem;
}
