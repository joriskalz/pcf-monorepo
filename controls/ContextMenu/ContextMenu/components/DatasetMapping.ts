import { CanvasContextMenuItem } from './Component.types';
import {
    ContextualMenuItemType,
    getColorFromString,
    getShade,
    IButtonStyles,
    IContextualMenuItem,
    IIconProps,
    Shade,
} from '@fluentui/react';
import { ItemColumns } from '../ManifestConstants';
const CSS_IMPORTANT = ' !important';
export function getMenuItemsFromDataset(dataset: ComponentFramework.PropertyTypes.DataSet): CanvasContextMenuItem[] {
    if (dataset.error || dataset.paging.totalResultCount === undefined) {
        // Dataset is not defined so return dummy items
        return [getDummyAction('1'), getDummyAction('2'), getDummyAction('3')];
    }
    const keyIndex: Record<string, number> = {};
    return dataset.sortedRecordIds.map((id) => {
        const record = dataset.records[id];
        // Prevent duplicate keys by appending the duplicate index
        let key = (record.getFormattedValue(ItemColumns.Key) as string) ?? '';
        if (keyIndex[key] !== undefined) {
            keyIndex[key]++;
            key += `_${keyIndex[key]}`;
        } else keyIndex[key] = 1;
        return {
            id: record.getRecordId(),
            key: key,
            name: record.getFormattedValue(ItemColumns.DisplayName),
            enabled: (record.getValue(ItemColumns.Enabled) as boolean) ?? undefined,
            iconName: record.getFormattedValue(ItemColumns.IconName),
            iconColor: record.getFormattedValue(ItemColumns.IconColor),
            parentItemKey: record.getFormattedValue(ItemColumns.ParentKey),
            iconOnly: (record.getValue(ItemColumns.IconOnly) as boolean) ?? undefined,
            checked: (record.getValue(ItemColumns.Checked) as boolean) ?? undefined,
            visible: (record.getValue(ItemColumns.Visible) as boolean) ?? undefined,
            isHeader: (record.getValue(ItemColumns.ItemHeader) as boolean) ?? undefined,
            topDivider: (record.getValue(ItemColumns.ItemTopDivider) as boolean) ?? undefined,
            divider: (record.getValue(ItemColumns.ItemDivider) as boolean) ?? undefined,
            data: record,
        } as CanvasContextMenuItem;
    });
}

function getDummyAction(key: string): CanvasContextMenuItem {
    return {
        id: key,
        key: key,
        name: 'Item ' + key,
        iconName: 'Unknown',
    } as CanvasContextMenuItem;
}

export function getCommandsWithChildren(
    items: CanvasContextMenuItem[],
    disabled: boolean,
    onClick: (ev?: React.MouseEvent<HTMLButtonElement>, item?: IContextualMenuItem | undefined) => boolean,
): IContextualMenuItem[] {
    return items
        .filter((i) => !i.parentItemKey && i.visible !== false)
        .map((i) => {
            return getCommandBarItemProps(items, i, disabled, onClick);
        });
}

export function getCommandBarItemProps(
    items: CanvasContextMenuItem[],
    item: CanvasContextMenuItem,
    disabled: boolean,
    onClick: (ev?: React.MouseEvent<HTMLButtonElement>, item?: IContextualMenuItem | undefined) => boolean,
): IContextualMenuItem {
    const subItems = items
        .filter((i) => i.parentItemKey === item.key && i !== item && i.visible !== false)
        .map((i) => {
            return getCommandBarItemProps(items, i, disabled, onClick);
        });
    const buttonStyles = {
        root: {
            background: 'rgba(255, 255, 255,0)',
        },
    } as IButtonStyles;

    const iconProps =
        item.iconName &&
        ({
            iconName: item.iconName,
        } as IIconProps);

    // If custom icon color, find the shade slots for the hover/pressed state
    if (iconProps && item.iconColor) {
        const iconColor = item.iconColor;
        iconProps.styles = { root: { color: item.iconColor + CSS_IMPORTANT } };

        // Attempt to set a darker shade for hover/pressed colors
        const primaryColor = getColorFromString(iconColor);
        if (primaryColor) {
            const iconColorDarker = getShade(primaryColor, Shade.Shade8);
            const secondaryColorStyle = { color: iconColorDarker?.str + CSS_IMPORTANT };
            buttonStyles.iconHovered = secondaryColorStyle;
            buttonStyles.iconPressed = secondaryColorStyle;
            buttonStyles.iconExpanded = secondaryColorStyle;
            buttonStyles.iconChecked = secondaryColorStyle;
            buttonStyles.iconExpandedHovered = secondaryColorStyle;
        }
    }

    const props = {
        key: item.key,
        text: item.name,
        disabled: item.enabled === false || disabled,
        onClick: onClick,
        iconOnly: item.iconOnly === true,
        iconProps: iconProps,
        canCheck: item.checked !== undefined,
        checked: item.checked,
        data: item,
        buttonStyles: buttonStyles,
    } as IContextualMenuItem;

    if (item.isHeader === true && subItems.length > 0) {
        // Semantic Section
        props.itemType = ContextualMenuItemType.Section;
        props.sectionProps = {
            title: item.name,
            topDivider: item.topDivider,
            bottomDivider: item.divider,
            items: subItems,
        };
    } else if (item.isHeader === true) {
        // Simple Section Break
        props.itemType = ContextualMenuItemType.Header;
    } else if (item.divider === true) {
        // Divider Break
        props.itemType = ContextualMenuItemType.Divider;
    } else {
        props.itemType = ContextualMenuItemType.Normal;
        props.subMenuProps =
            subItems.length > 0
                ? {
                      items: subItems,
                  }
                : undefined;
    }
    return props;
}
