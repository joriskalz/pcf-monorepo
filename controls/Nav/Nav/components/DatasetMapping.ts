import { NavItem } from './Component.types';
import { INavLink } from '@fluentui/react';
import { ItemColumns } from '../ManifestConstants';

export function getMenuItemsFromDataset(dataset: ComponentFramework.PropertyTypes.DataSet): NavItem[] {
    if (dataset.error || dataset.paging.totalResultCount === undefined) {
        // Dataset is not defined so return dummy items
        return getDummyAction();
    }
    return dataset.sortedRecordIds.map((id) => {
        const record = dataset.records[id];
        return {
            id: record.getRecordId(),
            key: record.getValue(ItemColumns.Key) as string,
            name: record.getValue(ItemColumns.DisplayName) as string,
            enabled: record.getValue(ItemColumns.Enabled) as boolean,
            iconName: record.getValue(ItemColumns.IconName) as string,
            iconColor: record.getValue(ItemColumns.IconColor) as string,
            parentItemKey: record.getValue(ItemColumns.ParentKey) as string,
            visible: record.getValue(ItemColumns.Visible) as boolean,
            isExpanded: record.getValue(ItemColumns.Expanded) as boolean,
            data: record,
        } as NavItem;
    });
}

/**
 * This method is primarily to structure the links as a tree to support Fluent UI Nav
 * @param items
 * @returns
 */
export function getTreeBasedGrpLinks(items: NavItem[]): NavItem[] {
    let root: NavItem[] = [];
    //Creating a custom rootID for NavGroupLink root node.
    const rootID = 'rootID'.concat(items[0].key, items[items.length - 1].key);
    items.map((item) => {
        // identify root and orphan items and replace their parentkey with rootID
        if (item.parentItemKey !== undefined) {
            if (
                items
                    .map((item) => {
                        return item.key;
                    })
                    .indexOf(item.parentItemKey) === -1
            ) {
                item.parentItemKey = rootID;
            }
        } else {
            item.parentItemKey = rootID;
        }
    });
    //add custom root to accomodate orphan items or links
    items.unshift({ id: rootID, name: rootID, key: rootID, parentItemKey: null } as unknown as NavItem);
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const keyMapping = items.reduce((prevGrp: any, currentGrp, i) => {
        prevGrp[currentGrp.key] = i;
        return prevGrp;
    }, {});
    items.map((grp) => {
        // Handle the root element
        if (grp.parentItemKey == null) {
            root = [grp] as unknown as NavItem[];
            return;
        }
        if (items[keyMapping[grp.key]].visible !== false) {
            const parentGrp = items[keyMapping[grp.parentItemKey]] as unknown as INavLink;
            parentGrp.links = [...(parentGrp.links || []), grp];
        }
    });
    return root;
}

function getDummyAction(): NavItem[] {
    return [
        {
            id: '1',
            key: '1',
            name: 'Item ' + 1,
        },
        {
            id: '2',
            key: '2',
            name: 'Item ' + 2,
            isExpanded: true,
        },
        {
            id: '3',
            key: '3',
            name: 'Item ' + 3,
            isExpanded: true,
        },
        {
            id: '4',
            key: '4',
            name: 'Item ' + 4,
            visible: false,
        },
        {
            id: '5',
            key: '5',
            name: 'Item ' + 5,
            parentItemKey: '3',
        },
        {
            id: '6',
            key: '6',
            name: 'Item ' + 6,
            parentItemKey: '2',
            isExpanded: false,
        },
        {
            id: '7',
            key: '7',
            name: 'Item ' + 7,
            parentItemKey: '6',
        },
        {
            id: '8',
            key: '8',
            name: 'Item ' + 8,
            parentItemKey: '7',
            isExpanded: true,
        },
        {
            id: '9',
            key: '9',
            name: 'Item ' + 9,
            parentItemKey: '8',
        },
        {
            id: '10',
            key: '10',
            name: 'Item ' + 10,
            parentItemKey: '9',
            isExpanded: true,
        },
        {
            id: '11',
            key: '11',
            name: 'Item ' + 11,
        },
        {
            id: '12',
            key: '12',
            name: 'Item ' + 12,
        },
        {
            id: '13',
            key: '13',
            name: 'Item ' + 13,
            parentItemKey: '12',
        },
        {
            id: '14',
            key: '14',
            name: 'Item ' + 14,
            parentItemKey: '12',
        },
        {
            id: '15',
            key: '15',
            name: 'Item ' + 12,
            parentItemKey: '15',
        },
    ] as unknown as NavItem[];
}
