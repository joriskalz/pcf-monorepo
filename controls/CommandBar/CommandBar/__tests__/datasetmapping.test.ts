import { getCommandsWithChildren, getMenuItemsFromDataset } from '../components/DatasetMapping';
import { ItemColumns } from '../ManifestConstants';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';

describe('DatasetMapping', () => {
    beforeEach(() => jest.clearAllMocks());

    it('returns correct props', () => {
        const items = [
            new MockEntityRecord('1', {
                [ItemColumns.Key]: 'commandOpen',
                [ItemColumns.DisplayName]: 'Open',
                [ItemColumns.IconName]: 'Open',
                [ItemColumns.IconColor]: 'blue',
                [ItemColumns.Enabled]: true,
                [ItemColumns.IconOnly]: false,
            }),
            new MockEntityRecord('2', {
                [ItemColumns.Key]: 'commandNew',
                [ItemColumns.DisplayName]: 'New',
                [ItemColumns.IconName]: 'New',
                [ItemColumns.IconColor]: 'blue',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
            }),
            new MockEntityRecord('3', {
                [ItemColumns.Key]: 'commandSave',
                [ItemColumns.DisplayName]: 'Save',
                [ItemColumns.IconName]: 'Save',
                [ItemColumns.IconColor]: 'green',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
                [ItemColumns.FarItem]: true,
            }),
            // Sub Items First Level
            new MockEntityRecord('5', {
                [ItemColumns.Key]: 'commandNewFrom',
                [ItemColumns.DisplayName]: 'NewFrom',
                [ItemColumns.IconName]: 'New',
                [ItemColumns.IconColor]: 'blue',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
                [ItemColumns.ParentKey]: 'commandNew',
            }),
            new MockEntityRecord('6', {
                [ItemColumns.Key]: 'commandSaveAndClose',
                [ItemColumns.DisplayName]: 'SaveAndClose',
                [ItemColumns.IconName]: 'Save',
                [ItemColumns.IconColor]: 'green',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
                [ItemColumns.FarItem]: true,
                [ItemColumns.ParentKey]: 'commandSave',
            }),
            // Sub Items Second Level
            new MockEntityRecord('7', {
                [ItemColumns.Key]: 'commandNewFromSub1',
                [ItemColumns.DisplayName]: 'NewFromSub1',
                [ItemColumns.IconName]: 'New',
                [ItemColumns.IconColor]: 'blue',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
                [ItemColumns.ParentKey]: 'commandNewFrom',
            }),
            new MockEntityRecord('8', {
                [ItemColumns.Key]: 'commandNewFromSub2',
                [ItemColumns.DisplayName]: 'NewFromSub2',
                [ItemColumns.IconName]: 'New',
                [ItemColumns.IconColor]: 'blue',
                [ItemColumns.Enabled]: false,
                [ItemColumns.IconOnly]: true,
                [ItemColumns.ParentKey]: 'commandNewFrom',
            }),
        ];

        const onClickEvent = jest.fn();
        const actions = getMenuItemsFromDataset(new MockDataSet(items));
        const props = getCommandsWithChildren(actions, false, onClickEvent);
        expect(props).toMatchSnapshot();
    });

    it('removes duplicate keys', () => {
        const itemsWithDuplicates = [
            new MockEntityRecord('1', {
                [ItemColumns.Key]: 'key1',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('2', {
                [ItemColumns.Key]: 'key1',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('3', {
                [ItemColumns.Key]: 'key1',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('4', {
                [ItemColumns.Key]: '2',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('5', {
                [ItemColumns.Key]: '2',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('6', {
                [ItemColumns.Key]: 'unique3',
                [ItemColumns.DisplayName]: 'Open',
            }),
        ];

        const onClickEvent = jest.fn();
        const actions = getMenuItemsFromDataset(new MockDataSet(itemsWithDuplicates));
        expect(actions.length).toBe(6);
        expect(actions[0].key).toBe('key1');
        expect(actions[1].key).toBe('key1_2');
        expect(actions[2].key).toBe('key1_3');
        expect(actions[3].key).toBe('2');
        expect(actions[4].key).toBe('2_2');
        expect(actions[5].key).toBe('unique3');

        const props = getCommandsWithChildren(actions, false, onClickEvent);
        expect(props.length).toBe(6);
        expect(props[0].key).toBe('key1');
        expect(props[1].key).toBe('key1_2');
        expect(props[2].key).toBe('key1_3');
        expect(props[3].key).toBe('2');
        expect(props[4].key).toBe('2_2');
        expect(props[5].key).toBe('unique3');
    });

    it('does not cause stack overflow with circular refs', () => {
        const itemsWithDuplicates = [
            new MockEntityRecord('1', {
                [ItemColumns.Key]: 'top',
                [ItemColumns.DisplayName]: 'Top',
                [ItemColumns.DisplayName]: 'top',
            }),
        ];

        const onClickEvent = jest.fn();
        const actions = getMenuItemsFromDataset(new MockDataSet(itemsWithDuplicates));
        getCommandsWithChildren(actions, false, onClickEvent);
    });
});
