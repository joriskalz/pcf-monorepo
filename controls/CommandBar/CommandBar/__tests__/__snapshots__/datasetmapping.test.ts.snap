// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetMapping returns correct props 1`] = `
Array [
  Object {
    "buttonStyles": Object {
      "root": Object {
        "background": "rgba(255, 255, 255,0)",
      },
    },
    "canCheck": false,
    "checked": undefined,
    "data": Object {
      "checked": undefined,
      "data": MockEntityRecord {
        "getNamedReference": [MockFunction],
        "id": "1",
        "values": Object {
          "ItemDisplayName": "Open",
          "ItemEnabled": true,
          "ItemIconColor": "blue",
          "ItemIconName": "Open",
          "ItemIconOnly": false,
          "ItemKey": "commandOpen",
        },
      },
      "divider": undefined,
      "enabled": true,
      "farItem": undefined,
      "iconColor": "blue",
      "iconName": "Open",
      "iconOnly": false,
      "id": "1",
      "isHeader": undefined,
      "key": "commandOpen",
      "name": "Open",
      "overflow": undefined,
      "parentItemKey": undefined,
      "split": undefined,
      "topDivider": undefined,
      "visible": undefined,
    },
    "disabled": false,
    "iconOnly": false,
    "iconProps": Object {
      "iconName": "Open",
      "styles": Object {
        "root": Object {
          "color": "blue !important",
        },
      },
    },
    "itemType": 0,
    "key": "commandOpen",
    "onClick": [MockFunction],
    "split": undefined,
    "subMenuProps": undefined,
    "text": "Open",
  },
  Object {
    "buttonStyles": Object {
      "root": Object {
        "background": "rgba(255, 255, 255,0)",
      },
    },
    "canCheck": false,
    "checked": undefined,
    "data": Object {
      "checked": undefined,
      "data": MockEntityRecord {
        "getNamedReference": [MockFunction],
        "id": "2",
        "values": Object {
          "ItemDisplayName": "New",
          "ItemEnabled": false,
          "ItemIconColor": "blue",
          "ItemIconName": "New",
          "ItemIconOnly": true,
          "ItemKey": "commandNew",
        },
      },
      "divider": undefined,
      "enabled": false,
      "farItem": undefined,
      "iconColor": "blue",
      "iconName": "New",
      "iconOnly": true,
      "id": "2",
      "isHeader": undefined,
      "key": "commandNew",
      "name": "New",
      "overflow": undefined,
      "parentItemKey": undefined,
      "split": undefined,
      "topDivider": undefined,
      "visible": undefined,
    },
    "disabled": true,
    "iconOnly": true,
    "iconProps": Object {
      "iconName": "New",
      "styles": Object {
        "root": Object {
          "color": "blue !important",
        },
      },
    },
    "itemType": 0,
    "key": "commandNew",
    "onClick": [MockFunction],
    "split": undefined,
    "subMenuProps": Object {
      "items": Array [
        Object {
          "buttonStyles": Object {
            "root": Object {
              "background": "rgba(255, 255, 255,0)",
            },
          },
          "canCheck": false,
          "checked": undefined,
          "data": Object {
            "checked": undefined,
            "data": MockEntityRecord {
              "getNamedReference": [MockFunction],
              "id": "5",
              "values": Object {
                "ItemDisplayName": "NewFrom",
                "ItemEnabled": false,
                "ItemIconColor": "blue",
                "ItemIconName": "New",
                "ItemIconOnly": true,
                "ItemKey": "commandNewFrom",
                "ItemParentKey": "commandNew",
              },
            },
            "divider": undefined,
            "enabled": false,
            "farItem": undefined,
            "iconColor": "blue",
            "iconName": "New",
            "iconOnly": true,
            "id": "5",
            "isHeader": undefined,
            "key": "commandNewFrom",
            "name": "NewFrom",
            "overflow": undefined,
            "parentItemKey": "commandNew",
            "split": undefined,
            "topDivider": undefined,
            "visible": undefined,
          },
          "disabled": true,
          "iconOnly": true,
          "iconProps": Object {
            "iconName": "New",
            "styles": Object {
              "root": Object {
                "color": "blue !important",
              },
            },
          },
          "itemType": 0,
          "key": "commandNewFrom",
          "onClick": [MockFunction],
          "split": undefined,
          "subMenuProps": Object {
            "items": Array [
              Object {
                "buttonStyles": Object {
                  "root": Object {
                    "background": "rgba(255, 255, 255,0)",
                  },
                },
                "canCheck": false,
                "checked": undefined,
                "data": Object {
                  "checked": undefined,
                  "data": MockEntityRecord {
                    "getNamedReference": [MockFunction],
                    "id": "7",
                    "values": Object {
                      "ItemDisplayName": "NewFromSub1",
                      "ItemEnabled": false,
                      "ItemIconColor": "blue",
                      "ItemIconName": "New",
                      "ItemIconOnly": true,
                      "ItemKey": "commandNewFromSub1",
                      "ItemParentKey": "commandNewFrom",
                    },
                  },
                  "divider": undefined,
                  "enabled": false,
                  "farItem": undefined,
                  "iconColor": "blue",
                  "iconName": "New",
                  "iconOnly": true,
                  "id": "7",
                  "isHeader": undefined,
                  "key": "commandNewFromSub1",
                  "name": "NewFromSub1",
                  "overflow": undefined,
                  "parentItemKey": "commandNewFrom",
                  "split": undefined,
                  "topDivider": undefined,
                  "visible": undefined,
                },
                "disabled": true,
                "iconOnly": true,
                "iconProps": Object {
                  "iconName": "New",
                  "styles": Object {
                    "root": Object {
                      "color": "blue !important",
                    },
                  },
                },
                "itemType": 0,
                "key": "commandNewFromSub1",
                "onClick": [MockFunction],
                "split": undefined,
                "subMenuProps": undefined,
                "text": "NewFromSub1",
              },
              Object {
                "buttonStyles": Object {
                  "root": Object {
                    "background": "rgba(255, 255, 255,0)",
                  },
                },
                "canCheck": false,
                "checked": undefined,
                "data": Object {
                  "checked": undefined,
                  "data": MockEntityRecord {
                    "getNamedReference": [MockFunction],
                    "id": "8",
                    "values": Object {
                      "ItemDisplayName": "NewFromSub2",
                      "ItemEnabled": false,
                      "ItemIconColor": "blue",
                      "ItemIconName": "New",
                      "ItemIconOnly": true,
                      "ItemKey": "commandNewFromSub2",
                      "ItemParentKey": "commandNewFrom",
                    },
                  },
                  "divider": undefined,
                  "enabled": false,
                  "farItem": undefined,
                  "iconColor": "blue",
                  "iconName": "New",
                  "iconOnly": true,
                  "id": "8",
                  "isHeader": undefined,
                  "key": "commandNewFromSub2",
                  "name": "NewFromSub2",
                  "overflow": undefined,
                  "parentItemKey": "commandNewFrom",
                  "split": undefined,
                  "topDivider": undefined,
                  "visible": undefined,
                },
                "disabled": true,
                "iconOnly": true,
                "iconProps": Object {
                  "iconName": "New",
                  "styles": Object {
                    "root": Object {
                      "color": "blue !important",
                    },
                  },
                },
                "itemType": 0,
                "key": "commandNewFromSub2",
                "onClick": [MockFunction],
                "split": undefined,
                "subMenuProps": undefined,
                "text": "NewFromSub2",
              },
            ],
          },
          "text": "NewFrom",
        },
      ],
    },
    "text": "New",
  },
  Object {
    "buttonStyles": Object {
      "root": Object {
        "background": "rgba(255, 255, 255,0)",
      },
    },
    "canCheck": false,
    "checked": undefined,
    "data": Object {
      "checked": undefined,
      "data": MockEntityRecord {
        "getNamedReference": [MockFunction],
        "id": "3",
        "values": Object {
          "ItemDisplayName": "Save",
          "ItemEnabled": false,
          "ItemFarItem": true,
          "ItemIconColor": "green",
          "ItemIconName": "Save",
          "ItemIconOnly": true,
          "ItemKey": "commandSave",
        },
      },
      "divider": undefined,
      "enabled": false,
      "farItem": true,
      "iconColor": "green",
      "iconName": "Save",
      "iconOnly": true,
      "id": "3",
      "isHeader": undefined,
      "key": "commandSave",
      "name": "Save",
      "overflow": undefined,
      "parentItemKey": undefined,
      "split": undefined,
      "topDivider": undefined,
      "visible": undefined,
    },
    "disabled": true,
    "iconOnly": true,
    "iconProps": Object {
      "iconName": "Save",
      "styles": Object {
        "root": Object {
          "color": "green !important",
        },
      },
    },
    "itemType": 0,
    "key": "commandSave",
    "onClick": [MockFunction],
    "split": undefined,
    "subMenuProps": Object {
      "items": Array [
        Object {
          "buttonStyles": Object {
            "root": Object {
              "background": "rgba(255, 255, 255,0)",
            },
          },
          "canCheck": false,
          "checked": undefined,
          "data": Object {
            "checked": undefined,
            "data": MockEntityRecord {
              "getNamedReference": [MockFunction],
              "id": "6",
              "values": Object {
                "ItemDisplayName": "SaveAndClose",
                "ItemEnabled": false,
                "ItemFarItem": true,
                "ItemIconColor": "green",
                "ItemIconName": "Save",
                "ItemIconOnly": true,
                "ItemKey": "commandSaveAndClose",
                "ItemParentKey": "commandSave",
              },
            },
            "divider": undefined,
            "enabled": false,
            "farItem": true,
            "iconColor": "green",
            "iconName": "Save",
            "iconOnly": true,
            "id": "6",
            "isHeader": undefined,
            "key": "commandSaveAndClose",
            "name": "SaveAndClose",
            "overflow": undefined,
            "parentItemKey": "commandSave",
            "split": undefined,
            "topDivider": undefined,
            "visible": undefined,
          },
          "disabled": true,
          "iconOnly": true,
          "iconProps": Object {
            "iconName": "Save",
            "styles": Object {
              "root": Object {
                "color": "green !important",
              },
            },
          },
          "itemType": 0,
          "key": "commandSaveAndClose",
          "onClick": [MockFunction],
          "split": undefined,
          "subMenuProps": undefined,
          "text": "SaveAndClose",
        },
      ],
    },
    "text": "Save",
  },
]
`;
